integrated_channels/integrated_channel/transmitters/course_metadata.py

```diff
@@ -16,85 +18,86 @@
-                error_message='',
-                enterprise_customer_uuid=self.enterprise_configuration.enterprise_customer.uuid
-            ).latest('created')
-        except ObjectDoesNotExist:
-            # This should happen if there have been 0 successful catalog transmissions for this enterprise
-            last_audit_summary = {}
-        else:
-            last_audit_summary = json.loads(last_catalog_transmission.audit_summary)
-
-        audit_summary = payload.resolve_removed_courses(last_audit_summary)
-
         total_transmitted = 0
         errors = []
         status_codes = []
-        for serialized_payload, length in payload.get_serialized_data_blocks():
-            status_code, body = self.transmit_block(serialized_payload)
+        for course_metadata in payload.export():
+            status_code, body = self.transmit_block(course_metadata)
             status_codes.append(str(status_code))
             error_message = body if status_code >= 400 else ''
             if error_message:
                 errors.append(error_message)
             else:
-                total_transmitted += length
+                total_transmitted += len(course_metadata)
 
         error_message = ', '.join(errors) if errors else ''
         code_string = ', '.join(status_codes)
 
-        catalog_transmission_audit = CatalogTransmissionAudit(
+        # pylint: disable=invalid-name
+        CatalogTransmissionAudit = apps.get_model('integrated_channel', 'CatalogTransmissionAudit')
+        try:
+            last_catalog_transmission = CatalogTransmissionAudit.objects.filter(
+                error_message='',
+                enterprise_customer_uuid=self.enterprise_configuration.enterprise_customer.uuid
+            ).latest('created')
+            last_audit_summary = json.loads(last_catalog_transmission.audit_summary)
+        except ObjectDoesNotExist:
+            last_audit_summary = {}
+
+        CatalogTransmissionAudit(
             enterprise_customer_uuid=self.enterprise_configuration.enterprise_customer.uuid,
             total_courses=len(payload.courses),
             status=code_string,
             error_message=error_message,
-            audit_summary=json.dumps(audit_summary),
-        )
+            audit_summary=json.dumps(payload.resolve_removed_courses(last_audit_summary)),
+        ).save()
+
+    def transmit_block(self, course_metadata):
+        """
+        Transmit a block of course metadata to the integrated channel.
+
+        Handle any request errors and logging requirements.
+
+        Args:
+            course_metadata (bytes): A set of bytes containing a page's worth of course metadata
 
-        catalog_transmission_audit.save()
-        return catalog_transmission_audit
+        Returns:
+            status_code (int): An integer status for the HTTP request
+            body (str): The SAP SuccessFactors server's response body
```

Comment:
@UmanShahzad This should no longer reference SAPSF.

Reply:
:+1: